packages/react-components/src/components/Picker/Picker.tsx

```diff
@@ -70,32 +78,96 @@ export const Picker: React.FC<IPickerProps> = ({
     }
   }, [isListOpen]);
 
-  const handleOnTriggerClick = () => {
-    if (disabled) {
+  React.useEffect(() => onSelect(selectedItems), [selectedItems]);
+
+  const handleOnTriggerClick = (e: React.MouseEvent | KeyboardEvent) => {
+    const target = e.target as Element;
+
+    if (disabled || target.getAttribute('data-dismiss-icon')) {
       return;
     }
 
-    return setIsListOpen((prev) => !prev);
+    setIsListOpen((prev) => !prev);
   };
 
   const handleOnClose = () => {
     setIsListOpen(false);
   };
 
   const handleOnSelect = (item: IPickerListItem) => {
+    if (!multiselect) {
+      setIsListOpen(false);
+      return setSelectedItems([item]);
+    }
+
+    const currentSelectedItems = selectedItems;
+
+    if (!currentSelectedItems) {
+      return setSelectedItems([item]);
+    }
+
+    const newSelectedItems = currentSelectedItems.includes(item)
+      ? currentSelectedItems.filter((selectedItem) => selectedItem !== item)
+      : currentSelectedItems.concat(item);
+
+    setSelectedItems(newSelectedItems);
+  };
+
+  const handleOnSelectAll = () => {
     setIsListOpen(false);
-    setSelectedItem(item);
-    onSelect(item);
+
+    const itemsToSelect = items.filter(
+      (item) => !item.disabled && !item.groupHeader && item.key !== 'select-all'
```

Comment:
Consider placing `select-all` in a variable

Reply:
Additionally, I'd think about a helper function `isItemSelectable` that would move this condition outside filter